core/src/main/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactory.java

```diff
@@ -133,15 +134,24 @@ public final void initialize(TelemetryConfiguration configuration) {
             setInstrumentationKey(applicationInsightsConfig, configuration);
 
             TelemetrySampler telemetrySampler = getSampler(applicationInsightsConfig.getSampler());
-            setChannel(applicationInsightsConfig.getChannel(), telemetrySampler, configuration);
-
+            boolean channelIsConfigured = setChannel(applicationInsightsConfig.getChannel(), telemetrySampler, configuration);
+            if (!channelIsConfigured) {
```

Comment:
True. Here's my rationale: Don't crash their app. They'll see that there's no telemetry. Turn on SDKLogger for details.

Reply:
Not crashing the app is a very good idea :)

Reply:
By breaking, I meant not crashing the app or throwing. Just leave rest of the processing by returning. Apologies if there was a confusion. Returning is very well possible I think. And also I think we should initialize logger first (infact I just want to get rid of enum type logger in very near term ;) )

Reply:
Previously, it would throw NPEs instead of slotting in a NopChannel. Just a constant stream of NPEs in the logs when you don't have it configured properly. I think NopChannel is a better solution.